lib/Firefox/FirefoxDriverService.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Facebook\WebDriver\Firefox;
+
+use Facebook\WebDriver\Remote\Service\DriverService;
+
+class FirefoxDriverService extends DriverService
+{
+    /**
+     * @var string Name of the environment variable storing the path to the driver binary
+     */
+    const WEBDRIVER_FIREFOX_DRIVER = 'WEBDRIVER_FIREFOX_DRIVER';
+    /**
+     * @var string Default executable used when no other is provided
+     * @internal
+     */
+    const DEFAULT_EXECUTABLE = 'geckodriver';
+
+    /**
+     * @return static
+     */
+    public static function createDefaultService()
+    {
+        $pathToExecutable = getenv(static::WEBDRIVER_FIREFOX_DRIVER);
+        if ($pathToExecutable === false || $pathToExecutable === '') {
+            $pathToExecutable = static::DEFAULT_EXECUTABLE;
+        }
+
+        $port = 9515; // TODO: Get another free port if the default port is used.
```

Comment:
getenv ?

Reply:
🤔 I see this static constructor as for "default" service, so it should not have any configuration IMHO. Maybe rather automatic free port detection should be added.

However if you want to override this default port settings, you can use the default constructor and set driver arguments as needed.